src/server/server.go

```diff
@@ -59,10 +65,64 @@ func PprofMem(w http.ResponseWriter, r *http.Request) {
         runtime.GC()
 	w.Header().Add("Content-Type", "application/octet-stream")
         if err := pprof.WriteHeapProfile(w); err != nil {
-            log.Fatal("could not write memory profile: ", err)
+	    	log.Fatal("could not write memory profile: ", err)
         }
 }
 
+// Starts CPU profiling
+func PprofCpuStart(w http.ResponseWriter, r *http.Request) {
+	// user error: double start (previous profiling not stopped yet)
+	if cpuTemp != nil {
+		log.Printf("Already started cpu profiling\n")
+		w.WriteHeader(409) // conflict
+		return
+	}
+
+	// fresh cpu profiling
+	temp, err := os.CreateTemp("", CPU_TEMP_PATTERN)
+	if err != nil {
+		log.Fatal("could not create cpu temporary file")
+	}
+
+	log.Printf("created a temporary file: %s", temp.Name())
+	cpuTemp = temp
+
+	if err := pprof.StartCPUProfile(temp); err != nil {
+		log.Fatal("could not start CPU profile: ", err)
+	}
+
+	log.Printf("Started cpu profiling\n")
+}
+
+// Stops CPU profiling, writes profiled data to response, and does cleanup
+func PprofCpuStop(w http.ResponseWriter, r *http.Request) {
+	// user error: should start cpu profiling first
+	if cpuTemp == nil {
+		log.Printf("should start cpu profile before stopping it\n")
+		w.WriteHeader(400) // bad request
+		return
+	}
+
+	// flush profile data to file
+	pprof.StopCPUProfile()
+	tempFilename := cpuTemp.Name()
+	cpuTemp.Close()
+	cpuTemp = nil
+	defer os.Remove(tempFilename) // deferred cleanup
+
+	log.Printf("Reading from %s\n", tempFilename)
+	buffer, err := ioutil.ReadFile(tempFilename)
+	if err != nil {
+		log.Fatal("could not read file: ", err)
```

Comment:
let's not have any log.Fatals or panics.  Just sending an error back for the user to see is right for this.